trezor-common/src/tests/mod.rs

```diff
@@ -0,0 +1,584 @@
+            }),
+        }
+    }
+}
+
+impl From<chain::TxOutput> for crate::TxOutput {
+    fn from(value: chain::TxOutput) -> Self {
+        match value {
+            chain::TxOutput::Transfer(value, dest) => Self::Transfer(value.into(), dest.into()),
+            chain::TxOutput::LockThenTransfer(value, dest, lock) => {
+                Self::LockThenTransfer(value.into(), dest.into(), lock.into())
+            }
+            chain::TxOutput::Burn(amount) => Self::Burn(amount.into()),
+            chain::TxOutput::DataDeposit(data) => Self::DataDeposit(data),
+            chain::TxOutput::CreateDelegationId(dest, pool_id) => {
+                Self::CreateDelegationId(dest.into(), pool_id.to_hash().into())
+            }
+            chain::TxOutput::DelegateStaking(amount, delegation_id) => {
+                Self::DelegateStaking(amount.into(), delegation_id.to_hash().into())
+            }
+            chain::TxOutput::ProduceBlockFromStake(dest, pool_id) => {
+                Self::ProduceBlockFromStake(dest.into(), pool_id.to_hash().into())
+            }
+            chain::TxOutput::Htlc(value, lock) => Self::Htlc(value.into(), (*lock).into()),
+            chain::TxOutput::CreateOrder(data) => Self::CreateOrder((*data).into()),
+            chain::TxOutput::CreateStakePool(pool_id, data) => {
+                Self::CreateStakePool(pool_id.to_hash().into(), (*data).into())
+            }
+            chain::TxOutput::IssueNft(token_id, data, dest) => {
+                Self::IssueNft(token_id.to_hash().into(), (*data).into(), dest.into())
+            }
+            chain::TxOutput::IssueFungibleToken(data) => Self::IssueFungibleToken((*data).into()),
+        }
+    }
+}
+
+fn make_random_destination(rng: &mut (impl Rng + CryptoRng)) -> chain::Destination {
+    match chain::DestinationTag::iter().choose(rng).unwrap() {
+        chain::DestinationTag::AnyoneCanSpend => chain::Destination::AnyoneCanSpend,
+        chain::DestinationTag::PublicKey => chain::Destination::PublicKey(
+            crypto::key::PrivateKey::new_from_rng(rng, KeyKind::Secp256k1Schnorr).1,
+        ),
+        chain::DestinationTag::ScriptHash => {
+            chain::Destination::ScriptHash(primitives::H256(rng.gen()).into())
+        }
+        chain::DestinationTag::ClassicMultisig => chain::Destination::ClassicMultisig(
+            (&crypto::key::PrivateKey::new_from_rng(rng, KeyKind::Secp256k1Schnorr).1).into(),
+        ),
+        chain::DestinationTag::PublicKeyHash => chain::Destination::PublicKeyHash(
+            (&crypto::key::PrivateKey::new_from_rng(rng, KeyKind::Secp256k1Schnorr).1).into(),
+        ),
+    }
+}
+
+fn make_random_bytes(rng: &mut (impl Rng + CryptoRng)) -> Vec<u8> {
+    (0..rng.gen_range(1..10)).map(|_| rng.gen()).collect()
+}
+
+fn make_random_account_command(rng: &mut (impl Rng + CryptoRng)) -> chain::AccountCommand {
+    match crate::AccountCommandIndex::iter().choose(rng).unwrap() {
```

Comment:
Btw, the idea was that you add Tag enum to `chain::AccountCommand` so that if it gets more variants, we get a compilation failure here. Same for other "XXXIndex" enums.

Also, I suggest calling them "XXXTag", because "index" is a bit confusing IMO. I.e. we also have those `#[codec(index = n)]` annotations on the "parent" enums, which may not be consecutive, and "XXXIndex" sounds like at has something to do with those indices. And "XXXTag" sounds like it's just a name.

Reply:
It is hard to re-export the AccountCommandTag from the original as it would bring a dependency on that crate, and it is not nostd compliant. As for the handling of new variants it will be caught by the From implementations, they do a match on the original enums.

I replaced Index with Type as it sounds better to me, AccountCommandType than Tag?

Reply:
> It is hard to re-export the AccountCommandTag from the original as it would bring a dependency on that crate, and it is not nostd compliant.

But it's for tests only, which can use std, right? I mean, you already use `chain::DestinationTag`, which also comes from a crate that is not nostd compliant.

> As for the handling of new variants it will be caught by the From implementations, they do a match on the original enums.

Yeah, I guess it will work. But IMO it'd still be a bit nicer if the match was over Tag types (or whatever they will be called) that come with the type that is being generated. Otherwise it's not obvious that the test actually generates all possible "chain::AccountCommand"s

> I replaced Index with Type as it sounds better to me, AccountCommandType than Tag?

Doesn't Type sound ambiguous?
In any case, we should probably use the same suffix for all such discriminant types. I.e. if "tag" is not good enough, `chain::DestinationTag` should be renamed too (I can do it myself or you can do it).
I'm not sure Type is a good alternative though. "Discriminant" might be better, but it's also longer.